diff -uNdr nut-0.45.0.old/Makefile.in nut-0.45.0/Makefile.in
--- nut-0.45.0.old/Makefile.in	Mon Mar 19 06:05:01 2001
+++ nut-0.45.0/Makefile.in	Mon Jul  2 12:11:29 2001
@@ -2,11 +2,10 @@
 
 # directory definitions
 prefix      = @prefix@
-BASEPATH    = $(INSTALLROOT)@prefix@
-CONFPATH    = $(INSTALLROOT)@sysconfdir@
-CGIPATH     = $(INSTALLROOT)@CGIPATH@
-INSTALLDIRS = $(CONFPATH) $(CGIPATH) $(MODELPATH) $(BASEPATH)/@bindir@  $(BASEPATH)/@sbindir@ $(BASEPATH)/misc
-STATEPATH   = $(INSTALLROOT)@STATEPATH@
+prefix      = @prefix@
+sysconfdir  = @sysconfdir@
+CGIPATH     = @CGIPATH@
+STATEPATH   = @STATEPATH@
 
 SUBDIRS	= common models server clients man
 CFLAGS	= -I../include @CFLAGS@
@@ -45,33 +44,22 @@
 		rm $$i/Makefile; \
 	done
 
-install: install-dirs
+install:
 	@echo Installing in $(BASEPATH)...
 
 	@for i in $(SUBDIRS) conf; do \
-		cd $$i; $(MAKE) install; cd ..; \
-	done
-
-install-dirs:
-	@for d in $(INSTALLDIRS); do \
-		if (test ! -d $$d) then \
-			$(INSTALLCMD) -d $$d || exit 1; \
-		fi \
+		cd $$i; $(MAKE) DESTDIR=$(DESTDIR) install; cd ..; \
 	done
 
-	@if (test ! -d $(STATEPATH)) then \
-		$(INSTALLCMD) -d -o $(RUNUID) -g $(RUNGID) $(STATEPATH) || exit 1; \
-	fi
-
 cgi:
 	@cd clients; $(MAKE) cgi; cd ..;
 
-install-cgi: install-dirs 
+install-cgi:
 	@cd conf; $(MAKE) install; cd ..;
-	@cd clients; $(MAKE) install-cgi; cd ..;
+	@cd clients; $(MAKE) DESTDIR=$(DESTDIR) install-cgi; cd ..;
 
-install-misc: install-dirs
-	@cd clients; $(MAKE) install-misc; cd ..;
+install-misc:
+	@cd clients; $(MAKE) DESTDIR=$(DESTDIR) install-misc; cd ..;
 
 # This is only used to set the "official" version before it goes out the door.
 #
diff -uNdr nut-0.45.0.old/clients/Makefile.in nut-0.45.0/clients/Makefile.in
--- nut-0.45.0.old/clients/Makefile.in	Thu Feb  1 06:02:53 2001
+++ nut-0.45.0/clients/Makefile.in	Mon Jul  2 12:11:29 2001
@@ -1,11 +1,10 @@
 # Network UPS Tools: clients
 
 # directories
-prefix	 = $(INSTALLROOT)@prefix@
-BASEPATH = $(prefix)
-CGIPATH	 = $(INSTALLROOT)@CGIPATH@
-BINPATH	 = $(prefix)/@bindir@
-SBINPATH = $(prefix)/@sbindir@
+prefix	 = @prefix@
+CGIPATH	 = @CGIPATH@
+bindir	 = @bindir@
+sbindir  = @sbindir@
 
 # other definitions
 CC	 = @CC@
@@ -66,18 +65,17 @@
 distclean: clean
 
 install: all
+	$(INSTALLCMD) -d $(DESTDIR)$(bindir)
+	$(INSTALLCMD) -d $(DESTDIR)$(sbindir)
 	@for f in $(PROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(BINPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(bindir); \
 	done
 	@for f in $(SPROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(SBINPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(sbindir); \
 	done
 
 install-cgi: cgi
+	$(INSTALLCMD) -d $(DESTDIR)$(CGIPATH)
 	@for f in $(CGIPROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(CGIPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(CGIPATH); \
 	done
-
-install-misc: upsfetch.o
-	$(INSTALLCMD) upsfetch.h $(BASEPATH)/misc
-	$(INSTALLCMD) upsfetch.o $(BASEPATH)/misc
diff -uNdr nut-0.45.0.old/conf/Makefile.in nut-0.45.0/conf/Makefile.in
--- nut-0.45.0.old/conf/Makefile.in	Fri Jun  1 11:23:15 2001
+++ nut-0.45.0/conf/Makefile.in	Mon Jul  2 12:28:20 2001
@@ -1,7 +1,7 @@
 # Network UPS Tools: conf
 
 prefix     = @prefix@
-CONFPATH   = $(INSTALLROOT)@sysconfdir@
+sysconfdir = @sysconfdir@
 INSTALLCMD = @INSTALL@
 
 SECFILES = upsmon.conf upsd.conf upsd.users
@@ -11,10 +11,11 @@
 clean:
 
 install:
+	$(INSTALLCMD) -d $(DESTDIR)$(sysconfdir)
 	@for f in $(SECFILES) ; do \
-		$(INSTALLCMD) -m 0600 $$f $(CONFPATH)/$$f.sample; \
+		$(INSTALLCMD) -m 0600 $$f $(DESTDIR)$(sysconfdir)/$$f; \
 	done
 
 	@for f in $(PUBFILES) ; do \
-		$(INSTALLCMD) -m 0644 $$f $(CONFPATH)/$$f.sample; \
+		$(INSTALLCMD) -m 0644 $$f $(DESTDIR)$(sysconfdir)/$$f; \
 	done
diff -uNdr nut-0.45.0.old/man/Makefile.in nut-0.45.0/man/Makefile.in
--- nut-0.45.0.old/man/Makefile.in	Tue Jan 23 22:01:15 2001
+++ nut-0.45.0/man/Makefile.in	Mon Jul  2 12:17:53 2001
@@ -1,7 +1,7 @@
 # Network UPS Tools: man
 
 prefix      = @prefix@
-MANDIR      = $(INSTALLROOT)@mandir@
+mandir      = @mandir@
 INSTALLCMD  = @INSTALL@
 INSTALLMODE = 0644
 
@@ -14,15 +14,7 @@
 distclean: clean
 
 install: all
-
-	@if (test ! -d $(MANDIR)) then \
-		$(INSTALLCMD) -d $(MANDIR) || exit 1;\
-	fi
-
-	@if (test ! -d $(MANDIR)/man8) then \
-		$(INSTALLCMD) -d $(MANDIR)/man8 || exit 1;\
-	fi
-
+	$(INSTALLCMD) -d $(DESTDIR)$(mandir)/man8
 	@for f in $(MAN8_PAGES) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(mandir)/man8; \
 	done
diff -uNdr nut-0.45.0.old/models/Makefile.in nut-0.45.0/models/Makefile.in
--- nut-0.45.0.old/models/Makefile.in	Sun May 27 08:56:01 2001
+++ nut-0.45.0/models/Makefile.in	Mon Jul  2 12:26:33 2001
@@ -1,7 +1,7 @@
 # Network UPS Tools: models
 
 # directories
-MODELPATH = $(INSTALLROOT)@MODELPATH@
+MODELPATH = @MODELPATH@
 
 # other definitions
 CC	= @CC@
@@ -54,10 +54,8 @@
 distclean: clean
 
 install: all
-	if (test ! -d $(MODELPATH)) then \
-		$(INSTALLCMD) -d $(MODELPATH) || exit 1; \
-	fi
+	$(INSTALLCMD) -d $(DESTDIR)$(MODELPATH)
 
 	@for f in $(PROGS) $(PROGS2) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MODELPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MODELPATH); \
 	done
diff -uNdr nut-0.45.0.old/server/Makefile.in nut-0.45.0/server/Makefile.in
--- nut-0.45.0.old/server/Makefile.in	Thu Mar 29 00:52:47 2001
+++ nut-0.45.0/server/Makefile.in	Mon Jul  2 12:20:57 2001
@@ -1,8 +1,8 @@
 # Network UPS Tools: server
 
 # directories
-prefix   = $(INSTALLROOT)@prefix@
-SBINPATH = $(prefix)/@sbindir@
+prefix   = @prefix@
+sbindir  = @sbindir@
 
 # other definitions
 CC	= @CC@
@@ -32,6 +32,7 @@
 distclean: clean
 
 install: all
+	$(INSTALLCMD) -d $(DESTDIR)$(sbindir)
 	@for f in $(PROGS) ; do \
-		$(INSTALLCMD) $$f $(SBINPATH); \
+		$(INSTALLCMD) $$f $(DESTDIR)$(sbindir); \
 	done
