--- nut-2.6.1/scripts/udev/nut-usbups.rules.in.orig	2011-06-01 16:03:34.000000000 +0200
+++ nut-2.6.1/scripts/udev/nut-usbups.rules.in	2011-07-26 13:03:40.529347458 +0200
@@ -67,6 +67,10 @@
 #  PW 9140  - usbhid-ups
 ATTR{idVendor}=="0592", ATTR{idProduct}=="0004", MODE="664", GROUP="@RUN_AS_GROUP@"
 
+# Orvaldi
+#  750/900SP - blazer_usb
+ATTR{idVendor}=="0665", ATTR{idProduct}=="5161", MODE="664", GROUP="@RUN_AS_GROUP@"
+
 # Phoenixtec
 #  various models  - bcmxcp_usb
 ATTR{idVendor}=="06da", ATTR{idProduct}=="0002", MODE="664", GROUP="@RUN_AS_GROUP@"
