diff -urNp -x '*.orig' nut-2.7.4.org/clients/Makefile.am nut-2.7.4/clients/Makefile.am
--- nut-2.7.4.org/clients/Makefile.am	2015-12-29 13:08:34.000000000 +0100
+++ nut-2.7.4/clients/Makefile.am	2021-10-05 21:27:39.927311204 +0200
@@ -18,7 +18,7 @@ if WITH_CGI
 endif
 
 bin_PROGRAMS = upsc upslog upsrw upscmd
-dist_bin_SCRIPTS = upssched-cmd
+dist_sbin_SCRIPTS = upssched-cmd
 sbin_PROGRAMS = upsmon upssched
 lib_LTLIBRARIES = libupsclient.la libnutclient.la
 if WITH_DEV
