diff -urN nut-0.99.1.org/Makefile.in nut-0.99.1/Makefile.in
--- nut-0.99.1.org/Makefile.in	Tue Aug 13 18:57:38 2002
+++ nut-0.99.1/Makefile.in	Tue Aug 13 19:31:19 2002
@@ -2,13 +2,12 @@
 
 # directory definitions
 prefix      = @prefix@
-INSTALLROOT = $(DESTDIR)
-BASEPATH    = $(INSTALLROOT)@prefix@
-CONFPATH    = $(INSTALLROOT)@sysconfdir@
-CGIPATH     = $(INSTALLROOT)@CGIPATH@
-DRVPATH     = $(INSTALLROOT)@DRVPATH@
-INSTALLDIRS = $(CONFPATH) $(DRVPATH) $(BASEPATH)/@bindir@ $(BASEPATH)/@sbindir@
-STATEPATH   = $(INSTALLROOT)@STATEPATH@
+BASEPATH    = @prefix@
+CONFPATH    = @sysconfdir@
+CGIPATH     = @CGIPATH@
+DRVPATH     = @DRVPATH@
+INSTALLDIRS = $(CONFPATH) $(DRVPATH) @bindir@ @sbindir@
+STATEPATH   = @STATEPATH@
 
 SUBDIRS	= common drivers server clients
 CFLAGS	= -I../include @CFLAGS@
@@ -44,7 +43,7 @@
 	done
 
 install: install-dirs install-man
-	@echo Installing in $(BASEPATH)...
+	@echo Installing in $(DESTDIR)$(BASEPATH)...
 
 	@for i in $(SUBDIRS) conf; do \
 		cd $$i; $(MAKE) install; cd ..; \
@@ -52,15 +51,15 @@
 
 install-dirs:
 	@for d in $(INSTALLDIRS); do \
-		if (test ! -d $$d) then \
-			./install-sh -d $$d || exit 1; \
+		if (test ! -d $(DESTDIR)$$d) then \
+			./install-sh -d $(DESTDIR)$$d || exit 1; \
 		fi \
 	done
 
-	@if (test ! -d $(STATEPATH)) then \
-		./install-sh -d -m $(STATEPATHMODE) $(STATEPATH) || exit 1 ; \
-		./install-sh -d -o $(RUNUID) -g $(RUNGID) $(STATEPATH) \
-		|| echo "Unable to set ownership of $(STATEPATH)"; \
+	@if (test ! -d $(DESTDIR)$(STATEPATH)) then \
+		./install-sh -d -m $(STATEPATHMODE) $(DESTDIR)$(STATEPATH) || exit 1 ; \
+		./install-sh -d -o $(RUNUID) -g $(RUNGID) $(DESTDIR)$(STATEPATH) \
+		|| echo "Unable to set ownership of $(DESTDIR)$(STATEPATH)"; \
 	fi
 
 install-man:
@@ -77,8 +76,8 @@
 	@cd clients; $(MAKE) cgi; cd ..;
 
 install-cgi: install-dirs install-cgi-man
-	if (test ! -d $(CGIPATH)) then \
-		./install-sh -d $(CGIPATH) || exit 1; \
+	if (test ! -d $(DESTDIR)$(CGIPATH)) then \
+		./install-sh -d $(DESTDIR)$(CGIPATH) || exit 1; \
 	fi \
 
 	@cd conf; $(MAKE) install; cd ..;
diff -urN nut-0.99.1.org/clients/Makefile.in nut-0.99.1/clients/Makefile.in
--- nut-0.99.1.org/clients/Makefile.in	Tue Aug 13 18:57:38 2002
+++ nut-0.99.1/clients/Makefile.in	Tue Aug 13 19:29:22 2002
@@ -1,12 +1,10 @@
 # Network UPS Tools: clients
 
 # directories
-INSTALLROOT = $(DESTDIR)
-prefix	 = $(INSTALLROOT)@prefix@
-BASEPATH = $(prefix)
-CGIPATH	 = $(INSTALLROOT)@CGIPATH@
-BINPATH	 = $(prefix)/@bindir@
-SBINPATH = $(prefix)/@sbindir@
+prefix	 = @prefix@
+CGIPATH	 = @CGIPATH@
+BINPATH	 = @bindir@
+SBINPATH = @sbindir@
 
 # other definitions
 CC	 = @CC@
@@ -69,20 +67,20 @@
 
 install: all
 	@for f in $(PROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(BINPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(BINPATH); \
 	done
 	@for f in $(SPROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(SBINPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(SBINPATH); \
 	done
 
 install-cgi: cgi
 	@for f in $(CGIPROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(CGIPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(CGIPATH); \
 	done
 
 install-misc: upsfetch.o
-	@if (test ! -d $(BASEPATH)/misc) then \
-		../install-sh -d $(BASEPATH)/misc || exit 1; \
+	@if (test ! -d $(prefix)/misc) then \
+		../install-sh -d $(prefix)/misc || exit 1; \
 	fi
-	$(INSTALLCMD) upsfetch.h $(BASEPATH)/misc
-	$(INSTALLCMD) upsfetch.o $(BASEPATH)/misc
+	$(INSTALLCMD) upsfetch.h $(prefix)/misc
+	$(INSTALLCMD) upsfetch.o $(prefix)/misc
diff -urN nut-0.99.1.org/conf/Makefile.in nut-0.99.1/conf/Makefile.in
--- nut-0.99.1.org/conf/Makefile.in	Tue Aug 13 18:57:38 2002
+++ nut-0.99.1/conf/Makefile.in	Tue Aug 13 19:29:22 2002
@@ -3,8 +3,7 @@
 # this gets used once configure expands sysconfdir below...
 prefix      = @prefix@
 
-INSTALLROOT = $(DESTDIR)
-CONFPATH    = $(INSTALLROOT)@sysconfdir@
+CONFPATH    = @sysconfdir@
 INSTALLCMD  = @INSTALL@
 
 SECFILES = upsmon.conf upsd.conf upsd.users
@@ -15,9 +14,9 @@
 
 install:
 	@for f in $(SECFILES) ; do \
-		$(INSTALLCMD) -m 0600 $$f $(CONFPATH)/$$f.sample; \
+		$(INSTALLCMD) -m 0600 $$f $(DESTDIR)$(CONFPATH)/$$f.sample; \
 	done
 
 	@for f in $(PUBFILES) ; do \
-		$(INSTALLCMD) -m 0644 $$f $(CONFPATH)/$$f.sample; \
+		$(INSTALLCMD) -m 0644 $$f $(DESTDIR)$(CONFPATH)/$$f.sample; \
 	done
diff -urN nut-0.99.1.org/drivers/Makefile.in nut-0.99.1/drivers/Makefile.in
--- nut-0.99.1.org/drivers/Makefile.in	Tue Aug 13 18:57:38 2002
+++ nut-0.99.1/drivers/Makefile.in	Tue Aug 13 19:29:22 2002
@@ -1,8 +1,7 @@
 # Network UPS Tools: drivers
 
 # directories
-INSTALLROOT = $(DESTDIR)
-DRVPATH = $(INSTALLROOT)@DRVPATH@
+DRVPATH = @DRVPATH@
 
 # other definitions
 CC	= @CC@
@@ -106,20 +105,20 @@
 # Install handlers
 
 install: all install-@DRIVER_INSTALL_TARGET@
-	$(INSTALLCMD) -m $(INSTALLMODE) upsdrvctl $(DRVPATH);
+	$(INSTALLCMD) -m $(INSTALLMODE) upsdrvctl $(DESTDIR)$(DRVPATH);
 
 # Install every driver
 
 install-all-drivers:
 	@for f in $(PROGS) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DRVPATH); \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DRVPATH); \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(DRVPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(DRVPATH); \
 	done
 
 # Just install the drivers that were selected by the user
 
 install-some-drivers:
 	@for f in $(DRIVER_BUILD_LIST) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DRVLPATH); \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DRVPATH); \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(DRVLPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(DRVPATH); \
 	done
diff -urN nut-0.99.1.org/man/Makefile.in nut-0.99.1/man/Makefile.in
--- nut-0.99.1.org/man/Makefile.in	Tue Aug 13 18:57:38 2002
+++ nut-0.99.1/man/Makefile.in	Tue Aug 13 19:29:22 2002
@@ -1,8 +1,7 @@
 # Network UPS Tools: man
 
-INSTALLROOT = $(DESTDIR)
 prefix      = @prefix@
-MANDIR      = $(INSTALLROOT)@mandir@
+MANDIR      = @mandir@
 INSTALLCMD  = @INSTALL@
 INSTALLMODE = 0644
 
@@ -34,35 +33,35 @@
 
 install: all install-man-dirs @DRIVER_INSTALL_TARGET@
 	@for f in $(MAN5_PAGES) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man5; \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man5; \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man5; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man5; \
 	done
 	@for f in $(MAN8_PAGES) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man8; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man8; \
 	done
 
 # make the directories for these things
 
 install-man-dirs:
 	@if (test ! -d $(MANDIR)) then \
-		echo ../install-sh -d $(MANDIR) || exit 1;\
-		../install-sh -d $(MANDIR) || exit 1;\
+		echo ../install-sh -d $(DESTDIR)$(MANDIR) || exit 1;\
+		../install-sh -d $(DESTDIR)$(MANDIR) || exit 1;\
 	fi
-	@if (test ! -d $(MANDIR)/man5) then \
-		echo ../install-sh -d $(MANDIR)/man5 || exit 1;\
-		../install-sh -d $(MANDIR)/man5 || exit 1;\
+	@if (test ! -d $(DESTDIR)$(MANDIR)/man5) then \
+		echo ../install-sh -d $(DESTDIR)$(MANDIR)/man5 || exit 1;\
+		../install-sh -d $(DESTDIR)$(MANDIR)/man5 || exit 1;\
 	fi
-	@if (test ! -d $(MANDIR)/man8) then \
-		echo ../install-sh -d $(MANDIR)/man8 || exit 1;\
-		../install-sh -d $(MANDIR)/man8 || exit 1;\
+	@if (test ! -d $(DESTDIR)$(MANDIR)/man8) then \
+		echo ../install-sh -d $(DESTDIR)$(MANDIR)/man8 || exit 1;\
+		../install-sh -d $(DESTDIR)$(MANDIR)/man8 || exit 1;\
 	fi
 
 # install every driver man page
 all-drivers:
 	@for f in $(DRIVERPAGES) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man8; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man8; \
 	done
 
 # install just the pages for the drivers that were built
@@ -70,19 +69,19 @@
 	@for f in $(DRIVER_BUILD_LIST) ; do \
 		if (test -f $$f.8) \
 		then \
-			echo $(INSTALLCMD) -m $(INSTALLMODE) $$f.8 $(MANDIR)/man8; \
-			$(INSTALLCMD) -m $(INSTALLMODE) $$f.8 $(MANDIR)/man8; \
+			echo $(INSTALLCMD) -m $(INSTALLMODE) $$f.8 $(DESTDIR)$(MANDIR)/man8; \
+			$(INSTALLCMD) -m $(INSTALLMODE) $$f.8 $(DESTDIR)$(MANDIR)/man8; \
 		fi \
 	done
 
 # just the CGI program man pages
 install-cgi-man: install-man-dirs
 	@for f in $(CGI_MAN5) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man5; \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man5; \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man5; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man5; \
 	done
 
 	@for f in $(CGI_MAN8) ; do \
-		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
+		echo $(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man8; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MANDIR)/man8; \
 	done
diff -urN nut-0.99.1.org/server/Makefile.in nut-0.99.1/server/Makefile.in
--- nut-0.99.1.org/server/Makefile.in	Tue Aug 13 18:57:38 2002
+++ nut-0.99.1/server/Makefile.in	Tue Aug 13 19:29:22 2002
@@ -1,9 +1,8 @@
 # Network UPS Tools: server
 
 # directories
-INSTALLROOT = $(DESTDIR)
-prefix   = $(INSTALLROOT)@prefix@
-SBINPATH = $(prefix)@sbindir@
+prefix   = @prefix@
+SBINPATH = @sbindir@
 
 # other definitions
 CC	= @CC@
@@ -31,6 +30,6 @@
 
 install: all
 	@for f in $(PROGS) ; do \
-		echo $(INSTALLCMD) $$f $(SBINPATH); \
-		$(INSTALLCMD) $$f $(SBINPATH); \
+		echo $(INSTALLCMD) $$f $(DESTDIR)$(SBINPATH); \
+		$(INSTALLCMD) $$f $(DESTDIR)$(SBINPATH); \
 	done
