diff -urN nut-1.1.14.org/clients/Makefile.in nut-1.1.14/clients/Makefile.in
--- nut-1.1.14.org/clients/Makefile.in	Wed Oct 30 17:05:02 2002
+++ nut-1.1.14/clients/Makefile.in	Wed Oct 30 17:05:23 2002
@@ -16,7 +16,7 @@
 CC	 = @CC@
 CFLAGS	 = -I../include @CFLAGS@
 PROGS	 = upsc upslog upsrw upscmd
-SPROGS	 = upsmon upssched
+SPROGS	 = upsmon upssched upssched-cmd
 CGIPROGS = upsstats.cgi upsimage.cgi upsset.cgi
 LIBDEP   = @LIBOBJ@
 LIBOBJ	 = $(LIBDEP) @NETLIBS@
