diff -Naur nut-0.45.2/configure.in nut-0.45.2-p/configure.in
--- nut-0.45.2/configure.in	Sun Sep  9 23:52:14 2001
+++ nut-0.45.2-p/configure.in	Thu Nov 22 12:38:00 2001
@@ -157,22 +157,7 @@
 	[ -lXpm ],
 )
 
-AC_CHECK_LIB(gd, gdImagePng,
-[
-	GFXLIBS="-lgd -lpng -lz"
-],
-[
-	echo "Trying again - using -L/usr/local/lib -I/usr/local/include..."
-	unset ac_cv_lib_gd_gdImagePng
-	CPPFLAGS="$CPPFLAGS -L/usr/local/lib -I/usr/local/include"
-	AC_CHECK_LIB(gd, gdImagePng,
-	[
-		GFXLIBS="-lgd -lpng -lz -L/usr/local/lib -I/usr/local/include"
-	],,
-	[-lm -lpng -lz])
-],
-[-lm -lpng -lz],
-)
+AC_CHECK_LIB(gd, gdImagePng, GFXLIBS="-lgd")
 
 if test "$GFXLIBS" = "" ; then
         echo "Trying again - using $LIBTTF"
