diff -urNp -x '*.orig' nut-2.7.4.org/drivers/apcsmart-old.h nut-2.7.4/drivers/apcsmart-old.h
--- nut-2.7.4.org/drivers/apcsmart-old.h	2021-10-05 21:27:40.590648101 +0200
+++ nut-2.7.4/drivers/apcsmart-old.h	2021-10-05 21:27:40.657315126 +0200
@@ -258,6 +258,7 @@ struct {
 	{ "0XM",	"789ABCDEFGKLMNOPQRSTUVWXYZcefgjklmnopqrsuwxz/<>\\^\014\026", 0 },
 	{ "0ZI",	"79ABCDEFGKLMNOPQRSUVWXYZcefgjklmnopqrsuxz/<>", 0 },
 	{ "5UI",	"79ABCDEFGKLMNOPQRSUVWXYZcefgjklmnopqrsuxz/<>", 0 },
+	{ "5YI",        "789ABCDEFGLMNOPQTUVWXYcefgjklmnopqrsuwxyz\\<>/", 0 },
 	{ "5ZM",	"79ABCDEFGKLMNOPQRSUVWXYZcefgjklmnopqrsuxz/<>", 0 },
 	/* APC600 */
 	{ "6QD",	"79ABCDEFGKLMNOPQRSUVWXYZcefgjklmnopqrsuxz", 0 },
