--- nut-2.4.3/m4/nut_check_libhal.m4~	2010-02-11 22:43:05.000000000 +0100
+++ nut-2.4.3/m4/nut_check_libhal.m4	2010-07-15 09:44:54.901127729 +0200
@@ -85,7 +85,19 @@
 		dnl Determine installation paths for callout and .fdi
 		dnl As per HAL spec, §5 Callouts and §2 Device Information Files
 		dnl - addon install path: $libdir/hal
+		HAL_CALLOUTS_PATH=""
 		AC_MSG_CHECKING(for libhal Callouts path)
+		AC_ARG_WITH(hal-callouts-path,
+		AC_HELP_STRING([--with-hal-callouts-path=PATH], [path for callout and .fdi files (auto)]),
+		[case "${withval}" in
+			yes|no)
+			;;
+			*)
+			HAL_CALLOUTS_PATH="${withval}"
+			;;
+		esac],)
+		if (test -z "${HAL_CALLOUTS_PATH}")
+		then                          
 		HAL_CALLOUTS_PATH=`pkg-config --silence-errors --variable=libexecdir hal`
 		if test -n "$HAL_CALLOUTS_PATH"; then
 			AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
@@ -109,9 +121,22 @@
 				AC_MSG_RESULT(using default (${HAL_CALLOUTS_PATH}))
 			fi
 		fi
+		fi
 
 		dnl - fdi install path: $datarootdir/hal/fdi/information/20thirdparty
+		HAL_FDI_PATH=""
 		AC_MSG_CHECKING(for libhal Device Information path)
+		AC_ARG_WITH(hal-fdi-path,
+		AC_HELP_STRING([--with-hal-fdi-path=PATH], [fdi install path (auto)]),
+		[case "${withval}" in
+			yes|no)
+			;;
+			*)
+			HAL_FDI_PATH="${withval}"
+			;;
+		esac],)
+		if (test -z "${HAL_FDI_PATH}")
+		then
 		HAL_FDI_PATH=`pkg-config --silence-errors --variable=hal_fdidir hal`
 		if test -n "$HAL_FDI_PATH"; then
 			HAL_FDI_PATH="${HAL_FDI_PATH}/information/20thirdparty"
@@ -121,6 +146,7 @@
 			HAL_FDI_PATH="${datarootdir}/hal/fdi/information/20thirdparty"
 			AC_MSG_RESULT(${HAL_FDI_PATH})
 		fi
+		fi
 	fi
 
 	CFLAGS="${CFLAGS_ORIG}"
