--- nut-0.44.1/conf/upsd.conf.orig	Thu Jul 20 21:22:13 2000
+++ nut-0.44.1/conf/upsd.conf	Sat Oct 21 17:03:57 2000
@@ -4,9 +4,9 @@
 # Having it owned by root, mode 0600 should be sufficient.
 
 # UPS <identifier> <statepath>/<model>-<port> [ <model> [<command line arguments>] <port> ]
-# UPS myups /var/state/ups/apcsmart-ttyS1 apcsmart /dev/ttyS1
-# UPS myups /var/state/ups/fentonups-tty00
-# UPS myups /var/state/ups/genericups-cuaa1 genericups -t 1 /dev/cuaa1
+# UPS myups /var/lib/ups/apcsmart-ttyS1 apcsmart /dev/ttyS1
+# UPS myups /var/lib/ups/fentonups-tty00
+# UPS myups /var/lib/ups/genericups-cuaa1 genericups -t 1 /dev/cuaa1
 #
 # Use this to specify which state files to monitor based on the model
 # support modules being run. The identifier isn't important unless 
