--- ../BUILD/nut-2.0.0/drivers/everups.c	2004-02-19 02:05:08.000000000 +0100
+++ nut-2.0.0/drivers/everups.c	2004-06-17 01:10:55.000000000 +0200
@@ -58,11 +58,15 @@
 {
         switch(upstype)
         {
-	        case 67: return "NET 500-DPC";
-		case 68: return "NET 700-DPC";
-		case 69: return "NET 1000-DPC";
-		case 70: return "NET 1400-DPC";
-		case 71: return "NET 2200-DPC";
+	        case 67: return "NET 500-DPC (old)";
+		case 68: return "NET 700-DPC (old)";
+		case 69: return "NET 1000-DPC (old)";
+		case 70: return "NET 1400-DPC (old)";
+		case 71: return "NET 2200-DPC (old)";
+ 		case 73: return "NET 700-DPC (new)";
+		case 74: return "NET 1000-DPC (new)";
+		case 75: return "NET 1400-DPC (new)";
+		case 76: return "NET 500-DPC (new)";
 		case 81: return "AP 450-PRO";
 		case 82: return "AP 650-PRO";
 		default:
@@ -109,7 +113,13 @@
 	if (Code(1)) {  /*Line voltage*/
 		ser_send_char(upsfd, 245);
 		ser_get_buf_len(upsfd, recBuf, 2, 3, 0);
+		
+	if ( upstype > 72 && upstype < 77)
+		lineV=(recBuf[0]*100+recBuf[1]*25600)/352;
+	else
 		lineV=(recBuf[0]*100+recBuf[1]*25600)/372;
+	
+		
 	} else {
 		upslog(LOG_INFO, "Code failed");
 		dstate_datastale();
