--- nut-2.7.4/configure.ac.orig	2017-12-03 20:59:45.766861382 +0100
+++ nut-2.7.4/configure.ac	2017-12-03 21:04:32.550191441 +0100
@@ -475,14 +475,7 @@
 if test "${nut_with_linux_i2c}" != no; then
     case ${target_os} in
         linux* )
-            AC_CHECK_DECLS(
-                [i2c_smbus_read_word_data, i2c_smbus_write_word_data, i2c_smbus_read_block_data],
-                [nut_with_linux_i2c="yes"],
-                [nut_with_linux_i2c="no"],
-                [#include <stdio.h>
-                 #include <linux/i2c-dev.h>
-                ]
-            )
+            AC_CHECK_LIB([i2c], [i2c_smbus_read_word_data], [nut_with_linux_i2c="yes"], [nut_with_linux_i2c="no"])
             ;;
         * )
             nut_with_linux_i2c="no"
--- nut-2.7.4/drivers/asem.c.orig	2015-12-29 13:08:34.000000000 +0100
+++ nut-2.7.4/drivers/asem.c	2017-12-03 21:08:50.156855165 +0100
@@ -35,6 +35,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <linux/i2c-dev.h>
+#include <i2c/smbus.h>
 
 #include "main.h"
 
--- nut-2.7.4/drivers/Makefile.am.orig	2016-03-08 13:01:11.000000000 +0100
+++ nut-2.7.4/drivers/Makefile.am	2017-12-03 21:16:08.973516822 +0100
@@ -225,7 +225,7 @@
 macosx_ups_SOURCES = macosx-ups.c
 
 # Asem
-asem_LDADD = $(LDADD_DRIVERS)
+asem_LDADD = $(LDADD_DRIVERS) -li2c
 asem_SOURCES = asem.c
 
 # nutdrv_qx USB/Serial
