diff -Naur nut-0.45.2/Makefile.in nut-0.45.2-p/Makefile.in
--- nut-0.45.2/Makefile.in	Thu Aug  9 18:52:31 2001
+++ nut-0.45.2-p/Makefile.in	Thu Nov 22 12:29:40 2001
@@ -2,11 +2,10 @@
 
 # directory definitions
 prefix      = @prefix@
-BASEPATH    = $(INSTALLROOT)@prefix@
-CONFPATH    = $(INSTALLROOT)@sysconfdir@
-CGIPATH     = $(INSTALLROOT)@CGIPATH@
-INSTALLDIRS = $(CONFPATH) $(CGIPATH) $(MODELPATH) $(BASEPATH)/@bindir@  $(BASEPATH)/@sbindir@ $(BASEPATH)/misc
-STATEPATH   = $(INSTALLROOT)@STATEPATH@
+prefix      = @prefix@
+sysconfdir  = @sysconfdir@
+CGIPATH     = @CGIPATH@
+STATEPATH   = @STATEPATH@
 
 SUBDIRS	= common models server clients man
 CFLAGS	= -I../include @CFLAGS@
@@ -44,7 +43,7 @@
 	@echo Installing in $(BASEPATH)...
 
 	@for i in $(SUBDIRS) conf; do \
-		cd $$i; $(MAKE) install; cd ..; \
+		cd $$i; $(MAKE) DESTDIR=$(DESTDIR) install; cd ..; \
 	done
 
 install-dirs:
@@ -55,18 +54,18 @@
 	done
 
 	@if (test ! -d $(STATEPATH)) then \
-		./install-sh -d -o $(RUNUID) -g $(RUNGID) $(STATEPATH) || exit 1; \
+		./install-sh -d $(DESTDIR)$(STATEPATH) || exit 1; \
 	fi
 
 cgi:
 	@cd clients; $(MAKE) cgi; cd ..;
 
 install-cgi: install-dirs 
-	@cd conf; $(MAKE) install; cd ..;
-	@cd clients; $(MAKE) install-cgi; cd ..;
+	@cd conf; $(MAKE) DESTDIR=$(DESTDIR) install; cd ..;
+	@cd clients; $(MAKE) DESTDIR=$(DESTDIR) install-cgi; cd ..;
 
 install-misc: install-dirs
-	@cd clients; $(MAKE) install-misc; cd ..;
+	@cd clients; $(MAKE) DESTDIR=$(DESTDIR) install-misc; cd ..;
 
 # This is only used to set the "official" version before it goes out the door.
 #
diff -Naur nut-0.45.2/clients/Makefile.in nut-0.45.2-p/clients/Makefile.in
--- nut-0.45.2/clients/Makefile.in	Wed Jul  4 06:31:29 2001
+++ nut-0.45.2-p/clients/Makefile.in	Thu Nov 22 12:29:40 2001
@@ -1,11 +1,10 @@
 # Network UPS Tools: clients
 
 # directories
-prefix	 = $(INSTALLROOT)@prefix@
-BASEPATH = $(prefix)
-CGIPATH	 = $(INSTALLROOT)@CGIPATH@
-BINPATH	 = $(prefix)/@bindir@
-SBINPATH = $(prefix)/@sbindir@
+prefix	 = @prefix@
+CGIPATH	 = @CGIPATH@
+bindir	 = @bindir@
+sbindir  = @sbindir@
 
 # other definitions
 CC	 = @CC@
@@ -67,18 +66,17 @@
 distclean: clean
 
 install: all
+	$(INSTALLCMD) -d $(DESTDIR)$(bindir)
+	$(INSTALLCMD) -d $(DESTDIR)$(sbindir)
 	@for f in $(PROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(BINPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(bindir); \
 	done
 	@for f in $(SPROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(SBINPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(sbindir); \
 	done
 
 install-cgi: cgi
+	$(INSTALLCMD) -d $(DESTDIR)$(CGIPATH)
 	@for f in $(CGIPROGS) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(CGIPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(CGIPATH); \
 	done
-
-install-misc: upsfetch.o
-	$(INSTALLCMD) upsfetch.h $(BASEPATH)/misc
-	$(INSTALLCMD) upsfetch.o $(BASEPATH)/misc
diff -Naur nut-0.45.2/conf/Makefile.in nut-0.45.2-p/conf/Makefile.in
--- nut-0.45.2/conf/Makefile.in	Mon Aug 13 19:12:18 2001
+++ nut-0.45.2-p/conf/Makefile.in	Thu Nov 22 12:29:40 2001
@@ -1,7 +1,7 @@
 # Network UPS Tools: conf
 
 prefix     = @prefix@
-CONFPATH   = $(INSTALLROOT)@sysconfdir@
+sysconfdir = @sysconfdir@
 INSTALLCMD = @INSTALL@
 
 SECFILES = upsmon.conf upsd.conf upsd.users
@@ -11,10 +11,11 @@
 clean:
 
 install:
+	$(INSTALLCMD) -d $(DESTDIR)$(sysconfdir)
 	@for f in $(SECFILES) ; do \
-		$(INSTALLCMD) -m 0600 $$f $(CONFPATH)/$$f.sample; \
+		$(INSTALLCMD) -m 0600 $$f $(DESTDIR)$(sysconfdir)/$$f; \
 	done
 
 	@for f in $(PUBFILES) ; do \
-		$(INSTALLCMD) -m 0644 $$f $(CONFPATH)/$$f.sample; \
+		$(INSTALLCMD) -m 0644 $$f $(DESTDIR)$(sysconfdir)/$$f; \
 	done
diff -Naur nut-0.45.2/man/Makefile.in nut-0.45.2-p/man/Makefile.in
--- nut-0.45.2/man/Makefile.in	Sat Oct 20 08:19:00 2001
+++ nut-0.45.2-p/man/Makefile.in	Thu Nov 22 12:32:44 2001
@@ -1,7 +1,7 @@
 # Network UPS Tools: man
 
 prefix      = @prefix@
-MANDIR      = $(INSTALLROOT)@mandir@
+mandir      = @mandir@
 INSTALLCMD  = @INSTALL@
 INSTALLMODE = 0644
 
@@ -17,18 +17,10 @@
 distclean: clean
 
 install: all
-	@if (test ! -d $(MANDIR)) then \
-		../install-sh -d $(MANDIR) || exit 1;\
-	fi
-	@if (test ! -d $(MANDIR)/man5) then \
-		../install-sh -d $(MANDIR)/man5 || exit 1;\
-	fi
-	@if (test ! -d $(MANDIR)/man8) then \
-		../install-sh -d $(MANDIR)/man8 || exit 1;\
-	fi
+	$(INSTALLCMD) -d $(DESTDIR)$(mandir)/{man5,man8}
 	@for f in $(MAN5_PAGES) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man5; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(mandir)/man5; \
 	done
 	@for f in $(MAN8_PAGES) ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MANDIR)/man8; \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(mandir)/man8; \
 	done
diff -Naur nut-0.45.2/models/Makefile.in nut-0.45.2-p/models/Makefile.in
--- nut-0.45.2/models/Makefile.in	Wed Aug 29 13:40:53 2001
+++ nut-0.45.2-p/models/Makefile.in	Thu Nov 22 12:29:41 2001
@@ -85,6 +85,7 @@
 distclean: clean
 
 install: all
+	$(INSTALLCMD) -d $(DESTDIR)$(MODELPATH)
 	@for f in $(PROGS) $(PROGS2) upsdrvctl ; do \
-		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(MODELPATH); \
+		$(INSTALLCMD) -m $(INSTALLMODE) $$f $(DESTDIR)$(MODELPATH); \
 	done
diff -Naur nut-0.45.2/server/Makefile.in nut-0.45.2-p/server/Makefile.in
--- nut-0.45.2/server/Makefile.in	Sun Jul 22 01:07:55 2001
+++ nut-0.45.2-p/server/Makefile.in	Thu Nov 22 12:29:41 2001
@@ -1,8 +1,8 @@
 # Network UPS Tools: server
 
 # directories
-prefix   = $(INSTALLROOT)@prefix@
-SBINPATH = $(prefix)/@sbindir@
+prefix   = @prefix@
+sbindir  = @sbindir@
 
 # other definitions
 CC	= @CC@
@@ -32,6 +32,7 @@
 distclean: clean
 
 install: all
+	$(INSTALLCMD) -d $(DESTDIR)$(sbindir)
 	@for f in $(PROGS) ; do \
-		$(INSTALLCMD) $$f $(SBINPATH); \
+		$(INSTALLCMD) $$f $(DESTDIR)$(sbindir); \
 	done
