--- nut-2.7.4/m4/nut_check_libopenssl.m4~	2015-12-29 09:42:34.000000000 +0100
+++ nut-2.7.4/m4/nut_check_libopenssl.m4	2018-09-18 08:24:37.208213839 +0200
@@ -58,7 +58,7 @@ if test -z "${nut_have_libopenssl_seen}"
 
 	dnl check if openssl is usable
 	AC_CHECK_HEADERS(openssl/ssl.h, [nut_have_openssl=yes], [nut_have_openssl=no], [AC_INCLUDES_DEFAULT])
-	AC_CHECK_FUNCS(SSL_library_init, [], [nut_have_openssl=no])
+	AC_CHECK_FUNCS(OPENSSL_init_ssl, [], [nut_have_openssl=no])
 
 	if test "${nut_have_openssl}" = "yes"; then
 		nut_with_ssl="yes"
--- nut-2.7.4/server/netssl.c~	2015-12-29 13:08:34.000000000 +0100
+++ nut-2.7.4/server/netssl.c	2018-09-18 08:26:46.435444419 +0200
@@ -387,7 +387,7 @@ void ssl_init(void)
 #ifdef WITH_OPENSSL
 
 	SSL_load_error_strings();
-	SSL_library_init();
+	OPENSSL_init_ssl(0, NULL);
 
 	if ((ssl_method = TLSv1_server_method()) == NULL) {
 		ssl_debug();
--- nut-2.7.4/clients/upsclient.c~	2015-12-29 13:08:34.000000000 +0100
+++ nut-2.7.4/clients/upsclient.c	2018-09-18 08:27:00.155858201 +0200
@@ -316,7 +316,7 @@ int upscli_init(int certverify, const ch
 	
 #ifdef WITH_OPENSSL
 	
-	SSL_library_init();
+	OPENSSL_init_ssl(0, NULL);
 	SSL_load_error_strings();
 
 	ssl_method = TLSv1_client_method();
