--- nut-2.6.4/scripts/systemd/nut-driver.service.in~	2012-05-02 23:35:36.000000000 +0200
+++ nut-2.6.4/scripts/systemd/nut-driver.service.in	2012-06-27 14:41:25.598754736 +0200
@@ -4,7 +4,7 @@
 StopWhenUnneeded=yes
 
 [Service]
-ExecStart=@SBINDIR@/upsdrvctl start
-ExecStop=@SBINDIR@/upsdrvctl stop
+ExecStart=@driverexecdir@/upsdrvctl start
+ExecStop=@driverexecdir@/upsdrvctl stop
 Type=forking
 
