--- nut-2.8.0/scripts/systemd/nut-server.service.in~	2022-04-23 13:56:07.000000000 +0200
+++ nut-2.8.0/scripts/systemd/nut-server.service.in	2022-07-22 12:36:28.363322701 +0200
@@ -19,10 +19,11 @@ PartOf=nut.target
 
 [Service]
 EnvironmentFile=-@CONFPATH@/nut.conf
+EnvironmentFile=-/etc/sysconfig/upsmon
 SyslogIdentifier=%N
 # Note: foreground mode by default skips writing a PID file (and
 # needs Type=simple); can use "-FF" here to create one anyway:
-ExecStart=@SBINDIR@/upsd -F
+ExecStart=@SBINDIR@/upsd -F $UPSD_OPTIONS
 ExecReload=@SBINDIR@/upsd -c reload -P $MAINPID
 
 [Install]
--- nut-2.8.0/scripts/systemd/nut-monitor.service.in~	2022-04-23 13:56:07.000000000 +0200
+++ nut-2.8.0/scripts/systemd/nut-monitor.service.in	2022-07-22 12:37:09.636656034 +0200
@@ -17,8 +17,9 @@ PartOf=nut.target
 
 [Service]
 EnvironmentFile=-@CONFPATH@/nut.conf
+EnvironmentFile=-/etc/sysconfig/upsmon
 SyslogIdentifier=%N
-ExecStart=@SBINDIR@/upsmon -F
+ExecStart=@SBINDIR@/upsmon -F $PROGRAM_ARGS
 ExecReload=@SBINDIR@/upsmon -c reload
 PIDFile=@PIDPATH@/upsmon.pid
 
